src/simple_commander/controllers/main.py

```diff
@@ -22,169 +26,189 @@
                   'invader': ['images/invader1.png', 'images/invader2.png']
                   }
 
+DEFAULT_SPEED = 5
+TIME_TO_SLEEP = 1  # 1 second, can be changed to 0.5
+g = 9.81
+
 
 class Unit(object):
 
-    def __init__(self, x, y, bonus, speed, unit_filename, bullet_filename):
+    def __init__(self, x, y, angle, bonus, speed, unit_filename, bullet_filename):
         self.image_filename = unit_filename
+        self.bullet_filename = bullet_filename
         self.x = x
         self.y = y
+        self.x0 = x
+        self.y0 = y
+        self.angle = angle
         self.width = 10  # temporary when we don't have real images
         self.height = 10  # must be height of real image
         self.bonus = bonus
         self.speed = speed
         self.is_dead = False
-        self.bullet = Bullet(self, bullet_filename)
+        self.step = 0
         self.shift = 5
 
     def to_json(self):
         return json.dumps(self, default=lambda o: o.__dict__)
 
+    def compute_new_coordinate(self, game_field):
+        max_height = game_field.get('height', 0)
+        max_width = game_field.get('width', 0)
+        self.step += TIME_TO_SLEEP
```

Comment:
I don't think we need to count step_duration. Let's shift x, y coordinates each iteration and it will always be x0, y0. Step duration in that case will be TIME_TO_SLEEP